Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java

```
@@ -237,4 +266,86 @@ public static String getPatientIdentifierIso(PatientType patient) {
+        throws GeneralSecurityException, IOException {
+        PrivateKey privkey = pair.getPrivate();
+        X509CertInfo info = new X509CertInfo();
+        Date fromDate = new Date();
+        Date toDate = new Date(fromDate.getTime() + days * 86400000l);
+        CertificateValidity interval = new CertificateValidity(fromDate, toDate);
+        BigInteger sn = new BigInteger(64, new SecureRandom());
+        X500Name owner = new X500Name(dn);
+
+        info.set(X509CertInfo.VALIDITY, interval);
+        info.set(X509CertInfo.SERIAL_NUMBER, new CertificateSerialNumber(sn));
+        info.set(X509CertInfo.SUBJECT, owner);
+        info.set(X509CertInfo.ISSUER, owner);
+        info.set(X509CertInfo.KEY, new CertificateX509Key(pair.getPublic()));
+        info.set(X509CertInfo.VERSION, new CertificateVersion(CertificateVersion.V3));
+        AlgorithmId algo = new AlgorithmId(AlgorithmId.md5WithRSAEncryption_oid);
+        info.set(X509CertInfo.ALGORITHM_ID, new CertificateAlgorithmId(algo));
+
+        // Sign the cert to identify the algorithm that's used.
+        X509CertImpl cert = new X509CertImpl(info);
+        cert.sign(privkey, algorithm);
+
+        // Update the algorith, and resign.
+        algo = (AlgorithmId) cert.get(X509CertImpl.SIG_ALG);
+        info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algo);
+        cert = new X509CertImpl(info);
+        cert.sign(privkey, algorithm);
+        return cert;
+    }
+
+    public static void saveJksTo(KeyStore keystore, String storePass, String storeLoc) {
+        try (FileOutputStream os = new FileOutputStream(storeLoc)) {
+            keystore.store(os, storePass.toCharArray());
+        } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException e) {
+            LOG.error("error unable to save to the keystore: {}", storeLoc, e.getLocalizedMessage(), e);
```

Should throw an exception and not fail silently.